src/sentry/hybridcloud/tasks/deliver_webhooks.py

```diff
@@ -77,25 +85,47 @@ def schedule_webhook_delivery(**kwargs: Never) -> None:
     Find mailboxes that contain undelivered webhooks that were scheduled
     to be delivered now or in the past.
 
+    Prioritizes webhooks based on provider importance.
+
     Triggered frequently by celery beat.
     """
-    # The double call to .values() ensures that the group by includes mailbox_nam
+    # The double call to .values() ensures that the group by includes mailbox_name
     # but only id_min is selected
     head_of_line = (
         WebhookPayload.objects.all()
         .values("mailbox_name")
         .annotate(id_min=Min("id"))
         .values("id_min")
     )
+
     # Get any heads that are scheduled to run
-    scheduled_mailboxes = WebhookPayload.objects.filter(
-        schedule_for__lte=timezone.now(),
-        id__in=Subquery(head_of_line),
-    ).values("id", "mailbox_name")
+    # Use provider field directly, with default priority for null values
+    scheduled_mailboxes = (
+        WebhookPayload.objects.filter(
+            schedule_for__lte=timezone.now(),
+            id__in=Subquery(head_of_line),
+        )
+        # Set priority value based on provider field
+        .annotate(
+            provider_priority=Case(
+                # For providers that match our priority list
+                *[
+                    When(provider=provider, then=Value(priority))
+                    for provider, priority in PROVIDER_PRIORITY.items()
+                ],
+                # Default value for all other cases (including null providers)
+                default=Value(DEFAULT_PROVIDER_PRIORITY),
+                output_field=CharField(),
+            )
+        )
+        # Order by priority first (lowest number = highest priority), then ID
+        .order_by("provider_priority", "id").values("id", "mailbox_name", "provider")
```

Comment:
Do you need to select `provider`? It isn't referenced in the results.

Reply:
I've removed the selection on `provider`.